src/libcore/iter.rs

```diff
@@ -3270,6 +3270,50 @@ impl<A, B> DoubleEndedIterator for Zip<A, B> where
 ///
 /// [`map()`]: trait.Iterator.html#method.map
 /// [`Iterator`]: trait.Iterator.html
+///
+/// # Notes about side effects
+///
+/// The [`map()`] iterator implements [`DoubleEndedIterator`], meaning that
+/// you can also [`map()`] backwards:
+///
+/// ```rust
+/// let v: Vec<i32> = vec![1, 2, 3].into_iter().rev().map(|x| x + 1).collect();
+///
+/// assert_eq!(v, [4, 3, 2]);
+/// ```
+///
+/// [`DoubleEndedIterator`]: trait.DoubleEndedIterator.html
+///
+/// However, if you call `.rev()` to iterate in the other direction, it
+/// will only be 'backwards' if your closure has no state. In other words,
+/// something like this might act in a way that you wouldn't expect:
+///
+/// ```rust
+/// // This iterator uses a closure that has side effects to count the number
+/// // of iterations.
+/// let counter = vec!['a', 'b', 'c'].into_iter().map({
+///     let mut count = 0;
+///     move |a| {
```

Comment:
Minor nit: using the variable name `a` when `'a'` is being iterated over is slightly confusing